SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java

```diff
@@ -998,49 +996,27 @@ public void visitSort(LogicalSort sort) {
         this.circuit.addOperator(agg);
 
         // Generate comparison function for sorting the vector
-        DBSPExpression comparators = null;
+        DBSPComparatorExpression comparator = new DBSPNoComparatorExpression(sort, inputRowType);
         for (RelFieldCollation collation: sort.getCollation().getFieldCollations()) {
             int field = collation.getFieldIndex();
             RelFieldCollation.Direction direction = collation.getDirection();
-            DBSPExpression comparator = new DBSPApplyExpression(
-                    DBSPTypeAny.instance.path(
-                            new DBSPPath("Extract", "new")),
-                    row.field(field).closure(row.asRefParameter()));
+            boolean ascending;
             switch (direction) {
                 case ASCENDING:
+                    ascending = true;
                     break;
                 case DESCENDING:
-                    comparator = new DBSPApplyMethodExpression("rev", DBSPTypeAny.instance, comparator);
+                    ascending = false;
                     break;
+                default:
                 case STRICTLY_ASCENDING:
                 case STRICTLY_DESCENDING:
                 case CLUSTERED:
                     throw new Unimplemented(sort);
             }
-            if (comparators == null)
-                comparators = comparator;
-            else
-                comparators = new DBSPApplyMethodExpression("then", DBSPTypeAny.instance, comparators, comparator);
+            comparator = new DBSPFieldComparatorExpression(sort, comparator, field, ascending);
```

Comment:
Why not just send the enum through rather a bool ascending=true/false? It's more informative than a boolean argument anyway.

Reply:
In principle the IR classes are independent of the Sql front-end classes, so they could work for other front-ends.
I also don't know the meaning of the other enumeration values in this case.